src/plugins/str.rs

```diff
@@ -33,6 +35,15 @@ impl Str {
         let applied = match self.action.as_ref() {
             Some(Action::Downcase) => Value::string(input.to_ascii_lowercase()),
             Some(Action::Upcase) => Value::string(input.to_ascii_uppercase()),
+            /*
+              This line was a test to see if anything is being pushed to self.params... it isn't
+              Some(Action::Substring) => Value::string(self.params[0].item.to_string()),
+              The following is just to avoid crashing on execution
+            */
+            Some(Action::Substring) => Value::string(input),
```

Comment:
Pattern matching lets you get the parameter for substring back out here, so you can use it to substr input